com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -260,16 +260,40 @@ internal string GenerateObserverErrorMessage(ClientRpcParams clientRpcParams, ul
         /// </summary>
         public NetworkManager NetworkManager => NetworkObject.NetworkManager;
 
+        private bool m_IsLocalPlayer;
         /// <summary>
         /// If a NetworkObject is assigned, it will return whether or not this NetworkObject
         /// is the local player object.  If no NetworkObject is assigned it will always return false.
         /// </summary>
-        public bool IsLocalPlayer { get; private set; }
+        public bool IsLocalPlayer
+        {
+            get
+            {
+                if (!NetworkObject.IsSpawned)
```

Comment:
I picked `IsSpawned` as it seems like the simple condition check that captured best what we need to check. Sensible?

Reply:
Also, the fact that we have both:
```
Runtime/Core/NetworkBehaviour.cs:311 :        public bool IsSpawned { get; internal set; }
Runtime/Core/NetworkObject.cs:96:        public bool IsSpawned { get; internal set; } (
```
is interesting. Sadly, the first one is not set by the time we run this check, so I rely on the other one. I'm not willing to investigate sweeping changes to have the two booleans change at the same time. :-)

Reply:
This was actually removed in [#1785](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/1785) to avoid exceptions under edge case scenarios and to help with performance.
**The general idea was this:**
If it is **not** spawned then those properties will be their default values.
If it **is** spawned then those values will be valid.

_We moved away from using other component references within setters and getters in order to avoid edge case scenarios where they might not be valid yet._

Awake and Start:
These can be used like they have been traditionally: to initialize "non-netcode aware" properties and/or components 
OnNetworkSpawn:
This should be used for initializing "netcode aware" components and at this time all "netcode aware" properties of NetworkBehaviour derived components will be valid.